lib/views/staging-view.js

```diff
@@ -391,20 +400,61 @@ export default class StagingView {
+      if (openNew) {
+        // User explicitly asked to view diff, such as via click
+        await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {activate: false});
+      } else {
+        const panesWithStaleItemsToUpdate = this.getPanesWithStalePendingFilePatchItem();
+        if (panesWithStaleItemsToUpdate.length > 0) {
+          // Update stale items to reflect new selection
+          await Promise.all(panesWithStaleItemsToUpdate.map(async pane => {
+            await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {
+              activate: false,
+              pane,
+            });
+          }));
+        } else {
+          // Selection was changed via keyboard navigation, update pending item in active pane
+          const activePane = this.props.workspace.getCenter().getActivePane();
+          const activePendingItem = activePane.getPendingItem();
+          const activePaneHasPendingFilePatchItem = activePendingItem && activePendingItem.getRealItem &&
+            activePendingItem.getRealItem() instanceof FilePatchController;
+          if (activePaneHasPendingFilePatchItem) {
+            await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {
+              activate: false,
+              pane: activePane,
+            });
+          }
+        }
       }
     }
   }
 
-  async showFilePatchItem(filePath, stagingStatus, {activate} = {activate: false}) {
+  getPanesWithStalePendingFilePatchItem() {
+    // "stale" meaning there is no longer a changed file associated with item
+    // due to changes being fully staged/unstaged/stashed/deleted/etc
+    return this.props.workspace.getPanes().filter(pane => {
+      const pendingItem = pane.getPendingItem();
+      if (!pendingItem) { return false; }
+      const isDiffViewItem = pendingItem.getRealItem && pendingItem.getRealItem() instanceof FilePatchController;
+      const isInActiveRepo = pendingItem.getWorkingDirectory() === this.props.workingDirectoryPath;
+      const isStale = !this.changedFileExists(pendingItem.getFilePath(), pendingItem.getStagingStatus());
+      return isDiffViewItem && isInActiveRepo && isStale;
+    });
+  }
+
+  changedFileExists(filePath, stagingStatus) {
+    return this.selection.findItem((item, key) => {
+      return key === stagingStatus && item.filePath === filePath;
+    });
+  }
+
+  async showFilePatchItem(filePath, stagingStatus, {activate, pane} = {activate: false}) {
     const encodedFilePath = encodeURIComponent(filePath);
     const encodedWorkdir = encodeURIComponent(this.props.workingDirectoryPath);
     const filePatchItem = await this.props.workspace.open(
       `atom-github://file-patch/${encodedFilePath}?workdir=${encodedWorkdir}&stagingStatus=${stagingStatus}`,
-      {pending: true, activatePane: activate, activateItem: activate},
+      {pending: true, activatePane: activate, activateItem: activate, pane},
```

Comment:
Ah, it'll be good to control the Pane here.

... I still want to do that change in :atom: core to be able to specify "categories" of pending pane, so that there's only one pending pane of each category at once and we can keep ours independent from tree-view's.